src/common/commonutils/UserUtils.c

```diff
@@ -3014,114 +2932,56 @@ int SetUsersRestrictedDotFiles(unsigned int* modes, unsigned int numberOfModes,
 
 int CheckUserAccountsNotFound(const char* names, char** reason, OsConfigLogHandle log)
 {
-    const char* userTemplate = "%s:";
-    size_t namesLength = 0;
-    char* name = NULL;
-    char* decoratedName = NULL;
-    SimplifiedUser* userList = NULL;
-    SimplifiedGroup* groupList = NULL;
-    unsigned int userListSize = 0, groupListSize = 0, i = 0, j = 0;
     bool found = false;
     int status = 0;
+    struct passwd* pw;
 
     if (NULL == names)
     {
         OsConfigLogError(log, "CheckUserAccountsNotFound: invalid argument");
         return EINVAL;
     }
 
-    namesLength = strlen(names);
-
-    if (0 == (status = EnumerateUsers(&userList, &userListSize, reason, log)))
+    setpwent();
+    for (errno = 0, pw = getpwent(); pw != NULL; errno = 0, pw = getpwent())
     {
-        for (i = 0; i < userListSize; i++)
-        {
-            for (j = 0; j < namesLength; j++)
-            {
-                if (NULL == (name = DuplicateString(&(names[j]))))
-                {
-                    OsConfigLogError(log, "CheckUserAccountsNotFound: failed to duplicate string");
-                    status = ENOMEM;
-                    break;
-                }
-                else
-                {
-                    TruncateAtFirst(name, ',');
-
-                    if (0 == strcmp(userList[i].username, name))
-                    {
-                        EnumerateUserGroups(&userList[i], &groupList, &groupListSize, reason, log);
-                        FreeGroupList(&groupList, groupListSize);
-
-                        OsConfigLogInfo(log, "CheckUserAccountsNotFound: user '%s' found with id %u, gid %u, home '%s' and present in %u group(s)",
-                            userList[i].username, userList[i].userId, userList[i].groupId, userList[i].home, groupListSize);
-
-                        if (DirectoryExists(userList[i].home))
-                        {
-                            OsConfigLogInfo(log, "CheckUserAccountsNotFound: home directory of user '%s' exists ('%s')", name, userList[i].home);
-                        }
-
-                        OsConfigCaptureReason(reason, "User '%s' found with id %u, gid %u, home '%s' and present in %u group(s)",
-                            userList[i].username, userList[i].userId, userList[i].groupId, userList[i].home, groupListSize);
-
-                        found = true;
-                    }
-                }
+        char* usernames = NULL;
+        char* token = NULL;
```

Comment:
We discussed to do this change longver term, but here let's  keep consistence with the rest of the file and pls move these up